src/blockcholesky.jl

```diff
@@ -1,27 +1,68 @@
-
-# Function for cholesky decomposition on block matries
-# 'cholesky' is the build-in one in LAPACK
-
-function Bcholesky(A::Symmetric{<:Any,<:BlockArray})
-    chol_U = BlockArray{Float64}(zeros(size(A)), fill(size(A)[1]÷blocksize(A)[1], blocksize(A)[1]), fill(size(A)[1]÷blocksize(A)[1], blocksize(A)[1]))
-
-    # Initializing the first role of blocks
-    chol_U[Block(1,1)] = cholesky!(A[Block(1,1)]).U
-    for j = 2:blocksize(A)[1]
-        chol_U[Block(1,j)] = chol_U[Block(1,1)]' \ A[Block(1,j)]
-    end
-
-    # For the left blocks
-    for i = 2:blocksize(A)[1]
-        for j = i:blocksize(A)[1]
-            if j == i
-                chol_U[Block(i,j)] = cholesky!(A[Block(i,j)] - sum(chol_U[Block(k,j)]'chol_U[Block(k,j)] for k in 1:j-1)).U
-            else
-                chol_U[Block(i,j)] = chol_U[Block(i,i)]' \ (A[Block(i,j)] - sum(chol_U[Block(k,i)]'chol_U[Block(k,j)] for k in 1:j-1))
-            end
-        end
-    end
-
-    return chol_U
-
-end
+
+##########################
+# Cholesky Factorization #
+##########################
+
+"""
+Instructions and examples
+"""
+
+"""
+Update
+1.changed all computations into in-place but there are still many allocations 
+which is highly depended on the number of blocks.
+For example, I took a block matrix A with n=5 and d=3 and it had 59 allocations. The build-in only have 5 instead.
+I will focus on reducing these allocations.

```

Comment:
Note Julia v1.4 allocates for things that shouldn't, for example subviews, so you won't ever get 0 allocations.

This has been improved in Julia v1.5, but expecting 0 allocations is unrealistic at this point.

Reply:
OK. I have played around with these today, finding that my function uses less memory but slower. Is this a trade-off or a result of inefficient algorithm I adopted?